baybe/searchspace/discrete.py

```diff
@@ -661,7 +662,35 @@ def transform(
         return comp_rep
 
 
-def _apply_constraint_filter(
+def _apply_polars_constraint_filter(
+    ldf: pl.LazyFrame, constraints: Collection[DiscreteConstraint]
+) -> pd.DataFrame:
+    """Remove discrete search space entries inplace based on constraints.
+
+    Args:
+        ldf: The data in experimental representation to be modified inplace.
+        constraints: List of discrete constraints.
+
+    Returns:
+        A Pandas Dataframe
+
+    """
+    # Limit constraints to Polars ones
+    constraints = [c for c in constraints if hasattr(c, "get_polars_expression")]
+    for c in constraints:
+        pl_expr = c.get_polars_expression(ldf)  # type: ignore[attr-defined]
+        if pl_expr is not None:
+            ldf = ldf.filter(pl_expr)
+
+    df = ldf.collect(streaming=True)
+    # Check if df is empty to prevent Polar's error
+    if df.shape[0] == 0:
+        return pd.DataFrame()
+
+    return df.to_pandas()
```

Comment:
Then I have to move the check in the above lines to `from_product` and before applying pandas filtering- because an attempt to convert empty Polars dataframe to Pandas will result in error.
Is it ok?

Reply:
i think o